src/services/formatting/formatting.ts

```diff
@@ -866,13 +866,23 @@ namespace ts.formatting {
             }
             else {
                 let tokenStart = sourceFile.getLineAndCharacterOfPosition(pos);
-                if (indentation !== tokenStart.character) {
-                    let startLinePosition = getStartPositionOfLine(tokenStart.line, sourceFile);
+                let startLinePosition = getStartPositionOfLine(tokenStart.line, sourceFile);
+                if (indentation !== tokenStart.character || indentationIsDifferent(indentationString, startLinePosition)) {

```

Comment:
why not just `indentationString === soruceFile.text.substr(startLinePosition , indentationString.length);`